app/src/main/java/org/mozilla/fenix/tabtray/TabsTouchHelper.kt

```diff
@@ -4,14 +4,94 @@
 
 package org.mozilla.fenix.tabtray
 
+import android.content.res.Resources
+import android.graphics.Canvas
+import android.graphics.drawable.Drawable
+import androidx.appcompat.content.res.AppCompatResources
 import androidx.recyclerview.widget.ItemTouchHelper
+import androidx.recyclerview.widget.RecyclerView
 import mozilla.components.browser.tabstray.TabTouchCallback
 import mozilla.components.concept.tabstray.TabsTray
 import mozilla.components.support.base.observer.Observable
+import mozilla.components.support.ktx.android.content.getColorFromAttr
+import mozilla.components.support.ktx.android.content.getDrawableWithTint
+import org.mozilla.fenix.R
+import org.mozilla.fenix.home.sessioncontrol.SwipeToDeleteCallback
+import org.mozilla.fenix.home.sessioncontrol.SwipeToDeleteCallback.Companion.DENSITY_CONVERSION
 
 class TabsTouchHelper(observable: Observable<TabsTray.Observer>) :
     ItemTouchHelper(object : TabTouchCallback(observable) {
-        override fun alphaForItemSwipe(dX: Float, distanceToAlphaMin: Int): Float {
-            return 1f - 2f * Math.abs(dX) / distanceToAlphaMin
+        override fun onChildDraw(
+            c: Canvas,
+            recyclerView: RecyclerView,
+            viewHolder: RecyclerView.ViewHolder,
+            dX: Float,
+            dY: Float,
+            actionState: Int,
+            isCurrentlyActive: Boolean
+        ) {
+            super.onChildDraw(c, recyclerView, viewHolder, dX, dY, actionState, isCurrentlyActive)
+            val icon = recyclerView.context.getDrawableWithTint(
+                R.drawable.ic_delete,
+                recyclerView.context.getColorFromAttr(R.attr.destructive)
+            )
+            val background =
+                AppCompatResources.getDrawable(recyclerView.context, R.drawable.tab_tray_background)
+            background?.let {
+                icon?.let {
+                    val itemView = viewHolder.itemView
+                    val iconLeft: Int
+                    val iconRight: Int
+                    val margin = convertDpToPixel(SwipeToDeleteCallback.MARGIN.toFloat())
```

Comment:
Can `dpToPx` be used here?
```suggestion
                    val margin = SwipeToDeleteCallback.MARGIN.dpToPx()
```